web/server/codechecker_server/database/db_cleanup.py

```diff
@@ -197,24 +197,37 @@ def upgrade_severity_levels(session_maker, checker_labels):
 
 def remove_unused_analysis_info(session_maker):
     """ Remove unused analysis information from the database. """
+    # Analysis info deletion is a relatively slow operation due to database
+    # cascades. Removing files in big chunks prevents reaching a potential
+    # database statement timeout. This hard-coded value is a safe choice
+    # according to some measurements.
+    CHUNK_SIZE = 500
```

Comment:
Is this batch size of 500 will completely resolve all database timeout during cleanup? If not maybe we should fall back to smaller size?